designer/client/src/helpers.js

```diff
@@ -9,3 +17,71 @@ export function arrayMove(arr, from, to) {
   arr.splice(to, 0, elm)
   return arr
 }
+
+/**
+ * Create filter for allowed page types
+ * @param {FormDefinition} data
+ * @param {Partial<Page>} page
+ */
+export function isControllerAllowed(data, page) {
+  const { pages } = data
+
+  // Check if we already have a start page
+  const hasStartPage = pages.some(({ controller }) => {
+    return controllerNameFromPath(controller) === ControllerType.Start
+  })
+
+  // Check if we already have a summary page
+  const hasSummaryPage = pages.some(({ controller }) => {
+    return controllerNameFromPath(controller) === ControllerType.Summary
+  })
+
+  // Check if we have a link from another page
+  const hasLinkFrom = findPathsTo(data, page.path).length > 1
+  const hasLinkTo = hasNext(page) && !!page.next.length
+
+  /**
+   * Filter allowed page types for current page
+   * @param {Page} pageType
+   */
+  return (pageType) => {
+    /**
+     * Start page unavailable when:
+     *
+     * 1. Another start page already exists
+     * 2. Current page is linked from another page
+     */
+    const isStartPageHidden =
+      pageType.controller === ControllerType.Start &&
+      (hasStartPage || hasLinkFrom)
+
+    /**
+     * Summary page unavailable when:
+     *
+     * 1. Another summary page already exists
+     * 2. Current page has links to another page
+     */
+    const isSummaryPageHidden =
+      pageType.controller === ControllerType.Summary &&
+      (hasSummaryPage || hasLinkTo)
+
+    /**
+     * Page types currently unavailable
+     */
+    const isInactivePage =
```

Comment:
What do you mean by "inactive" here?

Reply:
We now render the page type menu via [**controller-types.ts**](https://github.com/DEFRA/forms-designer/blob/main/model/src/pages/controller-types.ts) rather than hard coding the HTML

The following are "inactive" which means they don't appear in the UI:

1) File upload page
2) Status page (i.e. "Form submitted")

You'll see 1) is then enabled in PR #443 commit https://github.com/DEFRA/forms-designer/pull/443/commits/154b153a0243af00a4653831b727009460af75a4

We can enable 2) another day and let our content designers edit that page too

Reply:
Got it, thanks. Makes much more sense.